src/components/Popup.svelte

```diff
@@ -61,11 +61,18 @@
     justify-content: center;
     align-items: center;
   }
+
+  click-target {
+    position: fixed;
+    height: 100%;
+    width: 100%;
+  }
 </style>
 
+<click-target on:click|self={() => current && !dirty && dispatch('close')}></click-target>
 <box-container on:click|self="{() => current && !dirty && dispatch('close')}">
   <popup-box
-    on:click
+    on:click|stopPropagation
```

Comment:
what does this do?

Reply:
Events bubble up to parents by default, this prevents it from doing so, so we can tell if the container was clicked, or whether the popup itself was clicked